### PR TITLE
Adding endpoint parameters to multipart form data

### PR DESCRIPTION
**Pull request by syegin**
According to issue : #1137

### CODE REVIEW

File: Sources/Moya/MoyaProvider+Internal.swift [hunk 1]
```diff
@@ -291,6 +281,23 @@ private extension MoyaProvider {
     }
 }
 
+private extension Array where Element == MultipartFormData {
```
Comment (sunshinejr) [inline Sources/Moya/MoyaProvider+Internal.swift]: Maybe [MultipartFormData.swift](https://github.com/Moya/Moya/blob/master/Sources/Moya/MultipartFormData.swift) would be better for this extension?

**Comment by sunshinejr** [comment]
Hey @syegin, just checking if everything is alright. Let me know if you need some help or you just don't have time to do it right now.

**Comment by codwam** [comment]
Is this merge? I have the same problem.

**Comment by sunshinejr** [comment]
It is not yet merged, @codwam. I can take it from here (filling the tests & changelog), if @syegin won't in the near future.

**Comment by codwam** [comment]
@sunshinejr 
Hope you add it soon🍺

**Comment by dangthaison91** [comment]
@sunshinejr Could you help @syegin finish this PR and release it asap? We also have the same problem for several APIs.
At the moment, I have to manually create MultiFormData like PR in API Target.

**Comment by sunshinejr** [comment]
@dangthaison91 It might be the case that this PR won't make it into 9.0.0 release, since we are in the middle of changing the `parameters/encoding/Task` in #1147. Basically the API will be more verbose and you would specify what to do based on a task.

**Comment by sunshinejr** [comment]
This can be closed as #1147 is merged into 9.0.0-dev and will be into the new release soon. Thanks for your time, @syegin!

**Pull request closed without merging**